### ISSUE TITLE
[Feature Request] Add a "Disperse all siblings" button to the add-on

### ISSUE THREAD

**Issue by Expertium** [open-spaced-repetition/fsrs4anki#225]
**Which module is related to your feature request?**
Helper/Scheduler

**Is your feature request related to a problem? Please describe.**
The reason I want this feature is because when I go to Tools -> FSRS4Anki Helper -> Disperse Siblings in all decks, Anki just freezes. I have a large collection (150k cards total, 30k cards mature+learning), so performance issues aren't new for me, but this is the first time I'm having this big of a performance issue. I haven't tried waiting for hours for it to finish dispersing siblings and start responding again, and I probably won't try it - even if it does work, it's clearly too slow.

**Describe the solution you'd like**
When you click on the cog icon of a deck, you see buttons like "Postpone cards" and "Advance cards". I'd like to have "Disperse all siblings" there as well, it should work on a per-deck basis.

**Comment by L-M-Sherlock** [comment]
You can just use auto-rescheduling. The dispersing siblings would automatically work after you sync.

**Comment by Expertium** [comment]
You mean Tools -> FSRS4Anki Helper -> Auto reschedule recent reviews after sync? Yeah, it does work (well, with a bit of a delay, but at least without completely freezing), but what about cards that have siblings but haven't been recently reviewed?

**Comment by L-M-Sherlock** [comment]
> You mean Tools -> FSRS4Anki Helper -> Auto reschedule recent reviews after sync? Yeah, it does work (well, with a bit of a delay, but at least without completely freezing), but what about cards that have siblings but haven't been recently reviewed?

If these cards haven't been reviewed recently, why should we disperse them?

**Comment by Expertium** [comment]
If I understand how it works correctly, if only cards that have been recently reviewed will be dispersed, then it's possible to end up in a situation where you reviewed 2 (or more) siblings on the _same day_ and dispersed only _after_ reviewing all of them, which is suboptimal.

**Comment by user1823** [comment]
> then it's possible to end up in a situation where you reviewed 2 (or more) siblings on the same day and dispersed only after reviewing all of them

It won't occur if you enable the "Bury new siblings" and "Bury review siblings" options in the Deck Options of Anki.

**Comment by Expertium** [comment]
Ah, ok. Should I close this issue?

**Comment by user1823** [comment]
I think that a per-deck Disperse Siblings button is reasonable given that we have per-deck buttons for all the other features.

**Comment by L-M-Sherlock** [comment]
OK, I will add it to the development plan.

**Comment by L-M-Sherlock** [comment]
Patch:[fsrs4anki-helper.zip](https://github.com/open-spaced-repetition/fsrs4anki/files/11329566/fsrs4anki-helper.zip)
Commit: https://github.com/open-spaced-repetition/fsrs4anki-helper/commit/54a741f39e1076ddde459e30ddaff544df74738c

**Comment by L-M-Sherlock** [comment]
Does anybody want to test it?

**Comment by Expertium** [comment]
It seems to work fine for the most part, but there is one deck such that Anki freezes when I try to disperse siblings. I don't know if it has something to do with card type or with the amount of siblings (because in that deck almost every card has many siblings), I'm trying to figure it out right now.

**Comment by galantra** [comment]
I tried it on a collection which has 27,600 cards that are young or mature and not suspended. They belong to 8400 notes. 

I've been running it for 10 h already. No response from the program. Though it is probably just frozen, I left it running anyway in case it just takes a very long time.

**Comment by L-M-Sherlock** [comment]
Could you share your collection with me? I need to optimize the performance based on your cases.

**Comment by Expertium** [comment]
Here, I have included both the entire collection and the specific deck that causes the most trouble: https://drive.google.com/file/d/1NdIs58BVFFqM2RqE-OgC5mA5hlq-voUF/view?usp=share_link

**Comment by L-M-Sherlock** [comment]
Patch: [fsrs4anki-helper.zip](https://github.com/open-spaced-repetition/fsrs4anki/files/11359017/fsrs4anki-helper.zip)
Commit: https://github.com/open-spaced-repetition/fsrs4anki-helper/commit/b637c3376c398cb427dc2a2929c45b1567338253

**Comment by Expertium** [comment]
Much better! Now I can disperse all siblings in my collection without Anki freezing!
![image](https://user-images.githubusercontent.com/83031600/235300511-fcbf12e1-dd9e-4050-b609-99206271fe8c.png)

**Comment by L-M-Sherlock** [comment]
I find some notes have 10 or more cards, which freeze the Anki during dispersing. So I decrease the size of search states for these siblings. It could weaken the effect of dispersing.
